fahrschein/src/main/java/org/zalando/fahrschein/NakadiClient.java

```diff
@@ -109,8 +109,8 @@ public <T> void publish(String eventName, List<T> events) throws EventPublishing
             objectMapper.writeValue(body, events);
         }
 
-        eventPublishingHandlers.forEach(requestHandler -> requestHandler.onPublish(eventName, events));
         try (final Response response = request.execute()) {
+            eventPublishingHandlers.forEach(requestHandler -> requestHandler.onPublish(eventName, events));
```

Comment:
Now the `onPublish ` handler is executed after the request is executed ` try (final Response response = request.execute())`. is that ok ?

Reply:
> Now the `onPublish` handler is executed after the request is executed

Do you mean before? I read lines 114-117 to mean the sequence `onPublish`, `execute` (i.e. execute the request), `afterPublish`.

Reply:
It is fixed as part of this commit https://github.com/zalando-nakadi/fahrschein/pull/378/commits/c086a547d94095c8cae2ab48cee51a3c80d19ed3 👍